### PR TITLE
Int 971 fix default incident county

### PR DESCRIPTION
**Pull request by anuroop15**
### Jira Story

- [Default and lock county field in incident information card INT-971](https://osi-cwds.atlassian.net/browse/INT-971)

### Background
Tied up to original story: https://osi-cwds.atlassian.net/browse/INT-394

screen shot 2018-01-26 at 10 57 22 am

### CODE REVIEW

File: app/javascript/selectors/screening/incidentInformationFormSelector.js [hunk 1]
```diff
@@ -8,17 +7,14 @@ import {isFutureDatetimeCreate, combineCompact} from 'utils/validator'
...
-
 export const getIncidentDateSelector = createSelector(
   getIncidentInformationFormSelector,
   (incidentInformationForm) => incidentInformationForm.getIn(['incident_date', 'value'], '') || ''
 )
 
 export const getIncidentCountySelector = createSelector(
   getIncidentInformationFormSelector,
-  getDefaultIncidentCountySelector,
-  (incidentInformationForm, defaultIncidentCounty) => defaultIncidentCounty || incidentInformationForm.getIn(['incident_county', 'value'])
```
Comment (vinhboy) [inline app/javascript/selectors/screening/incidentInformationFormSelector.js]: Why is this here? It thought the revert should have removed this.

File: app/javascript/selectors/screening/incidentInformationFormSelector.js [hunk 2]
```diff
@@ -75,8 +74,8 @@ export const getVisibleErrorsSelector = createSelector(
   )
 )
 
-export const getCounties = () =>
-  Object.keys(COUNTIES).map((item) => ({key: item, name: COUNTIES[item]}))
+export const getCounties = (state) =>
+  (state.get('addressCounties').map((county) => ({key: county.get('code'), name: county.get('value')}))).toJS()
```
Comment (AmanShurafa) [inline app/javascript/selectors/screening/incidentInformationFormSelector.js]: This doesn't follow our selector naming convention. It wasn't truly a selector before as it wasn't selecting data from the store. But now it is.

File: spec/features/screening/create_screening_spec.rb [hunk 3]
```diff
@@ -123,6 +123,68 @@
...
+
+        within '.page-header-mast' do
+          expect(page).to have_content("Screening #{new_screening.id}")
+        end
+
+        expect(page).to have_select('Incident County', selected: 'Mendocino', disabled: :disabled)
+
+        within '#incident-information-card' do
+          click_button 'Cancel'
+          expect(page).to have_content('Mendocino')
```
[the remark this answers is not in the document]
Comment (anuroop15) [reply spec/features/screening/create_screening_spec.rb]: line 179 state the same assertion of having defaulted county in edit mode.

File: spec/javascripts/store/storeSpec.js [hunk 4]
```diff
@@ -33,6 +33,7 @@ describe('Store', () => {
       ethnicityTypes: [],
       languages: [],
       counties: [],
+      addressCounties: [],
```
[the remark this answers is not in the document]
Comment (anuroop15) [reply spec/javascripts/store/storeSpec.js]: added the commit to group into  reducers commit.

File: spec/features/screening/edit_screening_spec.rb [hunk 5]
```diff
@@ -22,7 +22,7 @@
       assignee: 'Bob Loblaw',
       communication_method: 'phone',
       ended_at: '2016-08-13T11:00:00.000Z',
-      incident_county: 'sacramento',
+      incident_county: '34',
```
[the remark this answers is not in the document]
Comment (anuroop15) [reply spec/features/screening/edit_screening_spec.rb]: I though of having the failing specs due controller as single commit. Will group with feature specs going forward

File: spec/javascripts/reducers/systemCodes/countyCodesReducerSpec.js [hunk 6]
```diff
@@ -6,7 +6,7 @@ import {fetchSuccess, fetchFailure} from 'actions/systemCodesActions'
 describe('countiesReducer', () => {
   beforeEach(() => jasmine.addMatchers(matchers))
 
-  describe('on FETCH_STATUS_CODES_COMPLETE', () => {
+  describe('on FETCH_SYSTEM_CODES_COMPLETE', () => {
```
[the remark this answers is not in the document]
Comment (anuroop15) [reply spec/javascripts/reducers/systemCodes/countyCodesReducerSpec.js]: Agreed. will follow from future commits.

File: spec/features/screening/create_screening_spec.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (AmanShurafa) [reply spec/features/screening/create_screening_spec.rb]: 👍

File: spec/javascripts/store/storeSpec.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (AmanShurafa) [reply spec/javascripts/store/storeSpec.js]: 👍

File: spec/features/screening/create_screening_spec.rb [hunk 7]
```diff
@@ -176,7 +176,7 @@
           expect(page).to have_content("Screening #{new_screening.id}")
         end
 
-        expect(page).to have_select('Incident County', selected: 'Mendocino', disabled: :disabled)
+        expect(page).to have_select('Incident County', selected: 'Mendocino', disabled: true)
```
Comment (AmanShurafa) [inline spec/features/screening/create_screening_spec.rb]: Next time, ensure the test fail properly before you make it green to avoid false positive again.
**Red, Green, Refactor**

File: spec/javascripts/views/IncidentInformationFormSpec.jsx [hunk 8]
```diff
@@ -87,6 +87,12 @@ describe('IncidentInformationForm', () => {
         })
         expect(component.find('SelectField[label="Incident County"]').props().value).toEqual('Yolo')
       })
+      it('disable dropdown when county exists', () => {
```
Comment (AmanShurafa) [inline spec/javascripts/views/IncidentInformationFormSpec.jsx]: `Incident County disable dropdown when county exists` doesn't read right. It also implies it's conditional and will only be disabled if county exists.

It should be `is disabled` so it would simply read `Incident County is disabled`

File: spec/javascripts/selectors/systemCodeSelectorsSpec.js [hunk 9]
```diff
@@ -29,3 +30,11 @@ describe('getOfficeLocationCodeValueSelector', () => {
     expect(getOfficeLocationCodeValueSelector(state)).toEqual(undefined)
   })
 })
+describe('getAddressCountiesSelector', () => {
+  it('return a list of address counties', () => {
+    const addressCounties = [{county_code: '99', value: 'State Of California'}]
+    const otherSystemCodes = [{county_code: '1', value: 'invalid'}]
+    const state = fromJS({addressCounties, otherSystemCodes})
+    expect(getAddressCountiesSelector(state).toJS()).toEqual(addressCounties)
+  })
```
Comment (AmanShurafa) [inline spec/javascripts/selectors/systemCodeSelectorsSpec.js]: Can you add a test to ensure it `returns a list when address counties are empty`?

File: app/javascript/selectors/screening/incidentInformationFormSelector.js (hunk 2 above)
Comment (AmanShurafa) [reply app/javascript/selectors/screening/incidentInformationFormSelector.js]: Ideally, we only add what we need. we don't need a list of all address counties right now since it would be defaulted. You only need the defaulted one. I realize that we might want the entire list in the future but that's a problem to be tackled in the future.

File: spec/javascripts/components/common/SelectFieldSpec.jsx [hunk 10]
```diff
@@ -85,5 +85,33 @@ describe('SelectField', () => {
...
+        id: 'myDateFieldId',
+        label: 'this is my label',
+        disabled: true,
+        value: 'this-is-my-value',
+        onChange: onChange,
+      }
+      component = shallow(<SelectField {...props}/>)
+      expect(component.find('select').prop('disabled')).toEqual(true)
+    })
+    it('renders an enabled select field when disabled prop set to false', () => {
```
Comment (AmanShurafa) [inline spec/javascripts/components/common/SelectFieldSpec.jsx]: `when disabled is not passed`

File: spec/javascripts/components/common/SelectFieldSpec.jsx [hunk 11]
```diff
@@ -85,5 +85,33 @@ describe('SelectField', () => {
...
+      const props = {
+        children: [],
+        gridClassName: 'myWrapperTest',
+        labelClassName: 'myLabelTest',
+        id: 'myDateFieldId',
+        label: 'this is my label',
+        disabled: false,
```
Comment (AmanShurafa) [inline spec/javascripts/components/common/SelectFieldSpec.jsx]: We either pass disabled or we don't , we should never hard code it to false.

Comment (anuroop15) [reply spec/javascripts/components/common/SelectFieldSpec.jsx]: we are not using default props here. so we need to pass hard coded value to unit test select dropdown field

Comment (AmanShurafa) [reply spec/javascripts/components/common/SelectFieldSpec.jsx]: No, it should pass even if you remove this line. That's how boolean props work they never need default props.

Comment (AmanShurafa) [reply spec/javascripts/components/common/SelectFieldSpec.jsx]: No, it should pass even if you remove this line. That's how boolean props work they never need default props.

File: spec/javascripts/components/common/SelectFieldSpec.jsx [hunk 12]
```diff
@@ -85,5 +85,33 @@ describe('SelectField', () => {
       expect(component.find('select').prop('required')).toEqual(true)
       expect(component.find('select').prop('aria-required')).toEqual(true)
     })
+    it('renders a disabled select field when disabled prop set to true', () => {
```
Comment (AmanShurafa) [inline spec/javascripts/components/common/SelectFieldSpec.jsx]: These 2 specs have nothing to do with required and shouldn't be under the same describe.

File: spec/javascripts/components/common/SelectFieldSpec.jsx [hunk 13]
```diff
@@ -85,5 +85,33 @@ describe('SelectField', () => {
...
+      const props = {
+        children: [],
+        gridClassName: 'myWrapperTest',
+        labelClassName: 'myLabelTest',
+        id: 'myDateFieldId',
+        label: 'this is my label',
+        disabled: true,
+        value: 'this-is-my-value',
+        onChange: onChange,
+      }
```
Comment (AmanShurafa) [inline spec/javascripts/components/common/SelectFieldSpec.jsx]: Defining `props` isn't necessary since it's already defined at the top level describe.

File: spec/javascripts/components/common/SelectFieldSpec.jsx [hunk 14]
```diff
@@ -85,5 +85,33 @@ describe('SelectField', () => {
...
+        gridClassName: 'myWrapperTest',
+        labelClassName: 'myLabelTest',
+        id: 'myDateFieldId',
+        label: 'this is my label',
+        disabled: false,
+        value: 'this-is-my-value',
+        onChange: onChange,
+      }
+      component = shallow(<SelectField {...props}/>)
+      expect(component.find('select').prop('disabled')).toEqual(false)
```
Comment (AmanShurafa) [inline spec/javascripts/components/common/SelectFieldSpec.jsx]: Try to read the attribute of the field instead of the prop. `.is('[disabled]')`

File: spec/javascripts/selectors/systemCodeSelectorsSpec.js [hunk 15]
```diff
@@ -29,3 +30,17 @@ describe('getOfficeLocationCodeValueSelector', () => {
...
+  it('return an empty list when address counties are empty', () => {
```
Comment (AmanShurafa) [inline spec/javascripts/selectors/systemCodeSelectorsSpec.js]: 👍

File: spec/javascripts/components/common/SelectFieldSpec.jsx [hunk 16]
```diff
@@ -69,21 +69,18 @@ describe('SelectField', () => {
...
-        gridClassName: 'myWrapperTest',
-        labelClassName: 'myLabelTest',
-        id: 'myDateFieldId',
-        label: 'this is my label',
-        required: true,
-        value: 'this-is-my-value',
-        onChange: onChange,
-      }
-      component = shallow(<SelectField {...props}/>)
+      component = shallow(<SelectField {...props} required/>)
```
Comment (AmanShurafa) [inline spec/javascripts/components/common/SelectFieldSpec.jsx]: 👍

File: app/javascript/selectors/screening/incidentInformationFormSelector.js (hunk 2 above)
Comment (VinhOSI) [reply app/javascript/selectors/screening/incidentInformationFormSelector.js]: I don't see a unit test for this?

**Comment by VinhOSI** [comment]
The commit message "add county_code to staff modal" should be "add county_code to staff model" right? I would fix that commit message because "modal" and "model" have two very different meanings.

**Comment by anuroop15** [comment]
screen shot 2018-01-30 at 11 48 00 am
